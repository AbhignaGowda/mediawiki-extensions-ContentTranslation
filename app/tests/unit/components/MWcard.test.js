import { shallowMount } from "@vue/test-utils";
import MwCard from "@/lib/mediawiki.ui/components/MWCard";

describe("MwCard.vue", () => {
  test("is a Vue instance", () => {
    const wrapper = shallowMount(MwCard);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders correctly", () => {
    const title = "Card title";
    const wrapper = shallowMount(MwCard, { propsData: { title } });
    expect(wrapper.find(".mw-ui-card__title").text()).toMatch(title);
    expect(wrapper.element).toMatchSnapshot();
  });
});
