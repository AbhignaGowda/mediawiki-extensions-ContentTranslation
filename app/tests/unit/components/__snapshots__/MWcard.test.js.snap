// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MwCard.vue renders correctly 1`] = `
<div
  class="mw-ui-card"
>
  <div
    class="card-title"
  >
    Card title
  </div>
   
</div>
`;
